test/metabase/test/async.clj

```diff
@@ -1,22 +1,10 @@
 (ns metabase.test.async
   "Utilities for testing async API endpoints."
-  (:require [clojure.tools.logging :as log]
-            [metabase.test.data.users :refer :all]))
+  (:require [metabase.feature-extraction.async :as async]
+            [metabase.models.computation-job :refer [ComputationJob]]))
 
-(def ^:private ^:const max-retries 20)
-
-(defn call-with-retries
-  "Retries fetching job results until `max-retries` times."
-  [user job-id]
-  (loop [tries 1]
-    (let [{:keys [status result] :as response}
-          ((user->client user) :get 200 (str "async/" job-id))]
-      (cond
-        (= status "done")     result
-        (= status "error")    (throw (ex-info (str "Error encountered.\n" result)
-                                       result))
-        (> tries max-retries) (throw (ex-info "Timeout. Max retries exceeded." {}))
-        :else                 (do
-                                (log/info (format "Waiting for computation to finish. Retry: %" tries))
-                                (Thread/sleep (* 100 tries))
-                                (recur (inc tries)))))))
+(defn result!
+  "Blocking version of async/result."
+  [job-id]
+  @((deref (deref #'async/running-jobs)) job-id)
```

Comment:
I had some difficult figuring out what's being checked here. `running-jobs` is an atom that holds a map. The key is a job-id, the value is a future? What's the third deref there for?

Reply:
The first one is for var, second for atom, third for future. I'll add a comment, and mixing @ and deref is gross.